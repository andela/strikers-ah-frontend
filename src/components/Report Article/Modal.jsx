import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faWindowClose } from '@fortawesome/fontawesome-free-solid';

const Modal = ({ handleClose, show, children }) => {
  const showHideClass = show ? 'modal display-block' : 'modal display-none';
  return (
    <div className={showHideClass}>
      <section className="modal-main">
        <button type="button" onClick={handleClose} className="closeModal">
          <FontAwesomeIcon icon={faWindowClose} size="2x" />
        </button>
        {children}
      </section>
    </div>
  );
};

// const container = document.createElement('div');
// document.body.appendChild(container);
export default Modal;
