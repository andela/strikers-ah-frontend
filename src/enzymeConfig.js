import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
