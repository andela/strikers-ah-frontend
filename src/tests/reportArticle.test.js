import React from 'react';
import { shallow } from 'enzyme';
import '../enzymeConfig';
import {
  ReportForm,
  mapStateToprops,
} from '../components/ReportArticle/ReportForm';
import Modal from '../components/ReportArticle/Modal';

const props = {
  article: {
    slug: 'fafasf',
  },
  report: {
    article: 'fasf',
  },
  reportCategory: {
    reportCategory: [
      { id: 1, name: 'fasklfjlak' },
      { id: 1, name: 'fasklfjlak' },
    ],
  },
  showModal: jest.fn(),
  createReport: jest.fn(),
  k: 1,
  state: {
    category: '1',
    description: 'Abuse',
    show: false,
  },
  getCategory: jest.fn(),
};

const modalProps = {
  handleClose: jest.fn(),
  children: jest.fn(),
  hideModal: jest.fn(),
  show: true,
};

const event = {
  target: { value: 'title', name: 'input0' },
};

const wrapper = shallow(<ReportForm {...props} />);
const Wrapper = shallow(<Modal {...modalProps} />);

describe('should test report form', () => {
  test('should render report form', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should render report form', () => {
    modalProps.show = true;
    expect(Wrapper).toMatchSnapshot();
  });

  test('should test the  `openModal` component', () => {
    const spy = jest.spyOn(wrapper.instance(), 'showModal');
    wrapper.instance().forceUpdate();
    const submitBtn = wrapper.find('#showmodal');
    submitBtn.simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test the  `reportButton` component', () => {
    const spy = jest.spyOn(wrapper.instance(), 'reportArticle');
    wrapper.instance().forceUpdate();
    const submitBtn = wrapper.find('#reportButton');
    submitBtn.simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  test('should test `mapStateToprops` in reportArcleDisplayMessage', () => {
    const mockState = {
      getCategory: {
        reportCategory: [],
      },
      Article: {
        article: {},
      },
    };
    const state = mapStateToprops(mockState);
    expect(state).toBeTruthy();
  });
  test('should test the `handleChange` method', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange');
    wrapper.instance().forceUpdate();
    const handlechange = wrapper.find('#test');
    handlechange.simulate('change', event);
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});
