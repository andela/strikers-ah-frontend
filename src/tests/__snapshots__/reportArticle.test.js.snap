// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test report form should render report form 1`] = `ShallowWrapper {}`;

exports[`should test report form should render report form 2`] = `ShallowWrapper {}`;
